java/src/jmri/managers/JmriUserPreferencesManager.java

```diff
@@ -403,6 +406,18 @@ public void setComboBoxLastSelection(String comboBoxName, String lastValue) {
         this.saveComboBoxLastSelections();
     }
 
+    @Override
+    public boolean getCheckboxPreferenceState(String name, boolean defaultState) {
+        return this.checkBoxLastSelection.getOrDefault(name, defaultState);
+    }
+
+    @Override
+    public void setCheckboxPreferenceState(String name, boolean state) {
+        checkBoxLastSelection.put(name, state);
+        setChangeMade(false);
```

Comment:
I don't know enough about this code to know whether it's a problem, but this uses `setChangeMade(false)` here and `resetChangeMade();` - should those be consistent?

Reply:
`setChangeMade()` sets the `dirty` flag. `resetChangeMade()` clears the `dirty` flag. The argument `false` to `setChangeMade()` only tells to not fire an update.
I have copied the code for ComboBox so they uses the same principle.